collections.go

```diff
@@ -71,30 +71,40 @@ func (dtos CollectionDTOs) ToShopify() shopify.Collections {
 
 // CollectionDTO represents a Shopify collection in HTTP requests and responses
 type CollectionDTO struct {
-	BodyHTML       string    `json:"body_html,omitempty"`
-	Handle         string    `json:"handle,omitempty"`
-	Image          ImageDTO  `json:"image,omitempty"`
-	ID             int64     `json:"id,omitempty"`
-	PublishedAt    time.Time `json:"published_at,omitempty"`
-	PublishedScope string    `json:"published_scope,omitempty"`
-	SortOrder      string    `json:"sort_order,omitempty"`
-	TemplateSuffix string    `json:"template_suffix,omitempty"`
-	Title          string    `json:"title,omitempty"`
-	UpdatedAt      time.Time `json:"updated_at,omitempty"`
+	BodyHTML       string     `json:"body_html,omitempty"`
+	Handle         string     `json:"handle,omitempty"`
+	Image          ImageDTO   `json:"image,omitempty"`
+	ID             int64      `json:"id,omitempty"`
+	PublishedAt    *time.Time `json:"published_at,omitempty"`
+	PublishedScope string     `json:"published_scope,omitempty"`
+	SortOrder      string     `json:"sort_order,omitempty"`
+	TemplateSuffix string     `json:"template_suffix,omitempty"`
+	Title          string     `json:"title,omitempty"`
+	UpdatedAt      *time.Time `json:"updated_at,omitempty"`
 }
 
 // ToShopify converts the DTO to the Shopify equivalent
 func (dto CollectionDTO) ToShopify() shopify.Collection {
+	var publishedAt time.Time
+	if !dto.PublishedAt.IsZero() {
+		publishedAt = *dto.PublishedAt
+	}
```

Comment:
You can't call `IsZero()` on `nil` https://play.golang.org/p/LyDuTTNKg3Q

If it's nil, you want a zeroed time. If it's not nil you want to deref the time.

Reply:
Looks like I can actually write tests in this package. I am going to add them in.

Reply:
yeah this should be
```go
if dto.PublishedAt != nil {
```